src/data/markdown/translated-guides/en/02 Using k6/14 Scenarios/00 About scenarios/02 VU allocation.md

```diff
@@ -0,0 +1,88 @@
+---
+title: VU allocation
+excerpt: How k6 allocates VUs in the open-model, arrival-rate executors
+---
+
+This document explains how k6 allocates VUs in the arrival-rate executors.
+
+In arrival-rate executors, three properties determine the iterations per second:
+-  `rate`  determines how many iterations k6 starts.
+- `timeUnit` determines how frequently it starts the number of iterations.
+- `preAllocatedVUs` sets the number of VUs to use to reach the target iterations per second.
+
+In short, while `rate` and `timeUnit` set the target iterations per second, **you must allocate enough VUs to reach this target.**
+By pre-allocating VUs, k6 can initialize the VUs necessary for the test before the test runs,
+which ensures that the CPU cost of allocation doesn't interfere with test execution.
```

Comment:
Some more explanation is probably needed here :thinking: Something like this?

```suggestion
The desired load in scenarios with arrival-rate executors is not configured in terms of how many VUs will loop through the iteration code in parallel. Instead, we need to specify at what interval k6 should try to start a new iteration, regardless of how many other iterations (and VUs) are currently running - the so-called ["open model"](/using-k6/scenarios/about-scenarios/open-vs-closed/#open-model). Or, put another way, we need to specify how many iterations per second (or per minute, hour, etc.) k6 should try and start.

Still, iterations need VUs to run on and k6 VUs, like other JavaScript runtimes, are single-threaded and can only run a single iteration (with its event loop) at any given time. This makes configuring scenarios with arrival-rate executors more complicated. We need to configure both the desired load, and to ensure we have enough pre-allocated initialized VUs to handle it. Which may be tricky to calculate, considering that the iteration duration depends entirely on what the script does and how quickly or slowly the system-under-test (SUT) responds. Slower SUT means slower iterations, which means more VUs will be needed, since some will be busy with running old iterations.

You can specify the desired arrival rate with the `rate` and `timeUnit` scenario options - k6 will try to start `rate` iterations evenly spread across a `timeUnit` (default `1s`) interval of time. So, given a scenario with a `constant-arrival-rate` executor and `rate: 10`, k6 will try to start one iteration every 100 milliseconds. However, if the scenario had `rate: 10, timeUnit: '1m'`, k6 will try to start a new iteration every 6 seconds.

VUs can be configured via the `preAllocatedVUs` option - it sets the number of VUs k6 will initialize before the test runs. There is also the `maxVUs` option, which allows k6 to initialize VUs mid-test, though that is mostly useful during development and generally is a bad practice to use in actual tests. VU initialization is somewhat CPU intensive and doing it in the middle of the test could potentially interfere with test execution or skew results.
```

Reply:
My initial thought is that the info is good...but dense.

One thing, I see that I was just wrong to type per second, it should be iteration rate.

 I also think a short bulleted intro should stay. Maybe not. This creates a [progressive disclosure](https://www.nngroup.com/articles/progressive-disclosure/), where readers who quickly scan could get the minimal info they need. Opening with four paragraphs in a row might scare many off.
 
 I've tried to synthesize your text and mine with:
 https://github.com/grafana/k6-docs/pull/974/commits/977c45d34e25c0defb079ef2c9f7bbb1e7279a24
 
 I wrote it sleepy, need to proofread, but let me know what you think.

Reply:
bullet points are fine, except when they force you to unnecessarily break up explanations so much that they stop making sense and are harder to understand, e.g. https://github.com/grafana/k6-docs/pull/974/files#r1069127202